definitions-tests/definitions_test.go

```diff
@@ -0,0 +1,74 @@
+package definitions_tests
+
+import (
+	"cuelang.org/go/cue"
+	"cuelang.org/go/cue/cuecontext"
+	"github.com/stretchr/testify/assert"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+const (
+	velaTemplatesDir   = "../vela-templates/definitions/"
+	testDataDir        = "./test-data/"
+	expectedOutputsKey = "outputs"
+	cueExtension       = ".cue"
+)
+
+func TestDefinitions(t *testing.T) {
+	// 1. traverse all definition .cue file
+	var targets []string
+	filepath.Walk(velaTemplatesDir, func(path string, info fs.FileInfo, err error) error {
+		if !info.IsDir() && strings.HasSuffix(info.Name(), cueExtension) {
+			targets = append(targets, strings.TrimPrefix(path, velaTemplatesDir))
+		}
+		return nil
+	})
+
+	// 2. lookup test data based on the same relative filepath
+	for _, targetPath := range targets {
+		testSubDir := filepath.Join(testDataDir, targetPath[:strings.LastIndex(targetPath, cueExtension)])
+		_, err := os.Stat(testSubDir)
+		if err != nil {
+			continue
+		}
+		entries, err := os.ReadDir(testSubDir)
+		if err != nil {
+			continue
+		}
+		for _, entry := range entries {
+			if !entry.IsDir() && strings.HasSuffix(entry.Name(), cueExtension) {
+				doTest(t, filepath.Join(testSubDir, entry.Name()), filepath.Join(velaTemplatesDir, targetPath))
+			}
+		}
+	}
+}
+
+func doTest(t *testing.T, testPath, targetPath string) {
+	// 1. parse test data
+	cuectx := cuecontext.New()
+	testdataBytes, err := os.ReadFile(testPath)
+	if err != nil {
+		assert.Errorf(t, err, "failed to read test .cue file %s", testPath)
+	}
+	testV := cuectx.CompileBytes(testdataBytes)
+	expectedOutputsV := testV.LookupPath(cue.ParsePath(expectedOutputsKey))
```

Comment:
if expectedOutputsV.Exist() {
   ...
}

Reply:
Added this assertation